tests/main/xdg-open-portal/task.yaml

```diff
@@ -0,0 +1,98 @@
+summary: Verify xdg-open forwarding requests to the desktop portal
+
+# Only enable the test on systems we know portals to function on.
+# Expand as needed.
+systems:
+  - ubuntu-18.04-*
+  - ubuntu-19.10-*
+
+environment:
+    BROWSER_HISTORY: /tmp/browser-history.txt
+    EDITOR_HISTORY: /tmp/editor-history.txt
+
+prepare: |
+    #shellcheck source=tests/lib/desktop-portal.sh
+    . "$TESTSLIB"/desktop-portal.sh
+    setup_portals
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+
+    install_local test-snapd-desktop
+
+    # Configure fake web browser
+    mkdir -p ~test/.local/share/applications
+    cat << EOF > ~test/.local/share/applications/test-browser.desktop
+    [Desktop Entry]
+    Type=Application
+    Name=Test Web Browser
+    Exec=$(pwd)/web-browser.sh %u $BROWSER_HISTORY
+    MimeType=x-scheme-handler/http;
+    EOF
+
+    # Configure a fake editor
+    cat << EOF > ~test/.local/share/applications/test-editor.desktop
+    [Desktop Entry]
+    Type=Application
+    Name=Test Editor
+    Exec=$(pwd)/editor.sh %f $EDITOR_HISTORY
+    MimeType=text/plain;
+    EOF
+
+    # Set up file type handlers
+    mkdir -p ~test/.config
+    cat << EOF > ~test/.config/mimeapps.list
+    [Default Applications]
+    x-scheme-handler/http=test-browser.desktop
+    text/plain=test-editor.desktop
+    EOF
+
+    # Prepare test data for editor
+    commondir=~test/snap/test-snapd-desktop/common
+    testfile="$commondir/test.txt"
+    mkdir -p "$commondir"
+    echo "Hello World" > "$testfile"
+    chown -R test:test ~test/snap
+
+restore: |
+    #shellcheck source=tests/lib/desktop-portal.sh
+    . "$TESTSLIB"/desktop-portal.sh
+    teardown_portals
+
+    rm -f ~test/.config/mimeapps.list
+    rm -f ~test/.local/share/applications/test-browser.desktop
+    rm -f ~test/.local/share/applications/test-editor.desktop
+    rm -f "$EDITOR_HISTORY"
+    rm -f "$BROWSER_HISTORY"
+
+execute: |
+    #shellcheck source=tests/lib/desktop-portal.sh
+    . "$TESTSLIB"/desktop-portal.sh
+    #shellcheck source=tests/lib/files.sh
+    . "$TESTSLIB"/files.sh
+
+    echo "Request is forwarded to the default browser"
+    as_user test-snapd-desktop.cmd xdg-open http://www.example.org
```

Comment:
Please consider switching to `session-tool`, which handles this part better IMO. I will try to remove `as_user` entirely from the test tree but this will take some time.

Reply:
When I looked at it, it appeared to spawn a new session for each invocation.  That's not really what I'd want for most testing of desktop or user session related features.

For these kinds of tests, I would like to start a user session, and then issue multiple commands (possibly with some privileged ones interspersed).